lib/puppet/resource/status.rb

```diff
@@ -88,6 +88,14 @@ class Status
       #   dependencies of this resource that failed to apply.
       attr_accessor :failed_dependencies
 
+      # @!attribute [rw] corrective_change
+      #   @return [Boolean] true if the resource contained a corrective change.
+      attr_accessor :corrective_change
```

Comment:
Per the other comments, I don't think that this should be writable since it seems like the event itself should be calculating this.